pkg/parse/run.go

```diff
@@ -218,6 +218,13 @@ func (r *reconciler) fetch(ctx context.Context) (*SourceStatus, cmpath.Absolute,
 	if newSourceStatus.Errs == nil {
```

Comment:
Do we need to remove the `ready-to-render` file when there is source error?

Reply:
Added logic to remove this file when a source error occurs, either before or after image verification. If the removal fails, it logs the error without appending it to the source error or interrupting any loop. Since the hydration controller won’t receive the expected commit in the ready-to-render file when a source error exists, rendering is effectively blocked regardless.